validator/slashing-protection/proposer_protect.go

```diff
@@ -0,0 +1,49 @@
+package slashingprotection
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/helpers"
+	"github.com/prysmaticlabs/prysm/shared/params"
+)
+
+// IsSlashableBlock checks if signed beacon block is slashable against
+// a validator's slashing protection history and against a remote slashing protector if enabled.
+func (s *Service) IsSlashableBlock(
+	ctx context.Context, block *ethpb.SignedBeaconBlock, pubKey [48]byte, domain *ethpb.DomainResponse,
+) (bool, error) {
+	if block == nil || block.Block == nil {
+		return false, errors.New("received nil block")
+	}
+	metricsKey := fmt.Sprintf("%#x", pubKey)
+	existingSigningRoot, err := s.validatorDB.ProposalHistoryForSlot(ctx, pubKey[:], block.Block.Slot)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to get proposal history")
+	}
+	// Check if the block is slashable by local and remote slashing protection
+	if s.remoteProtector != nil {
+		slashable, err := s.remoteProtector.IsSlashableBlock(ctx, block, pubKey, domain)
+		if err != nil {
+			return false, errors.Wrap(err, "failed to get proposal history")
+		}
+		remoteSlashableProposalsTotal.WithLabelValues(metricsKey).Inc()
+		return slashable, nil
+	}
```

Comment:
I am concerned that our local protection DB will be inadequate in this paradigm. 

Imagine a scenario where User is running a remote slasher, then they turn it off for whatever reason and restart their node without the remote slashing protection. 
Now they have an empty local slashing protection DB and are at higher risk for slashing.  

Recommend: When enabled, check remote slashing protection. Save results in local slashing DB always.